### ISSUE TITLE
WriteTag reports violation with confusing message when there is no javadoc

### ISSUE THREAD

**Issue by strkkk** [checkstyle/checkstyle#11584]
https://checkstyle.org/config_javadoc.html#WriteTag
```
$ cat Test.java
public class Test {}

$ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WriteTag"/>
    </module>
</module>

$ java -jar checkstyle-10.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\workdir\Test.java:1: Type Javadoc comment is missing null tag. [WriteTag]
Audit done.
Checkstyle ends with 1 errors.
```

Expected: no violation, since there is no javadoc.
Also note confusing violation message ("null" tag).

For javadoc presence there are other checks. And docs mention that with default config it does nothing, but it reports weird message instead.

P.S. Checkstyle has `WriteTag` with default config - 
https://github.com/checkstyle/checkstyle/blob/4ad253653d59d77002ae2f61349624dcb18dcc38/config/checkstyle_checks.xml#L653

--------

**Breaking compatibility migration notes:**
after fix and release (10.21.0) Check will not report violations on missed Javadocs.
To archive behavior as before, users should start using [InvalidJavadocPosition](https://checkstyle.org/checks/javadoc/invalidjavadocposition.html), [MissingJavadocMethod](https://checkstyle.org/checks/javadoc/missingjavadocmethod.html) , [MissingJavadocPackage](https://checkstyle.org/checks/javadoc/missingjavadocpackage.html) , [MissingJavadocType](https://checkstyle.org/checks/javadoc/missingjavadoctype.html) to let Checkstyle demand javadoc first and when WriteTag will  demand required tag in existing javadoc

**Comment by theWolf1198** [comment]
I want to work on this, however, I have a question:

This seems like an issue with a confusing message that is printed when there is no default tag, do you want me to refactor the output log message or change the check? 

Thanks!

**Comment by strkkk** [comment]
@theWolf1198 I expected check to do not report violation if there is no javadoc

**Comment by theWolf1198** [comment]
@romani @strkkk, I think that the code should allow empty Javadoc for default WriteTag as there are other checks to check if Javadoc is present or not. 
Please let me know your thoughts.

**Comment by theWolf1198** [comment]
@strkkk, sure I will start working on this right away.

**Comment by theWolf1198** [comment]
@strkkk , new PR raised for this issue, please check.

**Comment by coderdeadpool** [comment]
@romani Can i work on this issue.

**Comment by nrmancuso** [comment]
@coderdeadpool we have a PR open for this that is nearly complete, please take another issue. If you have done a gsoc first and second issue, it might be good to try to resolve some pitest suppressions: https://github.com/checkstyle/checkstyle/issues/12341

**Comment by romani** [comment]
yes, please takes , there is almost done PR, please reuse that code.
